File: src/main/java/com/nuvora/backend_finanzas/service/LeccionService.java
package com.nuvora.backend_finanzas.service;

import com.nuvora.backend_finanzas.dto.LeccionDTO;
import com.nuvora.backend_finanzas.entity.Leccion;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface LeccionService {

    LeccionDTO createLeccion (LeccionDTO leccionDTO);

    List<LeccionDTO> listLeccionByModulo(Long moduloId);

    LeccionDTO getLeccionById(Long leccionId);

    LeccionDTO updateLeccion(Long leccionId, LeccionDTO leccionDTO);

    void deleteLeccion(Long leccionId);

    Leccion getLeccionEntityById(Long leccionId);

}

File: frontend/src/app/services/leccion.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Leccion } from '../models/leccion';
import { API_BASE_URL } from '../config/api.config';

@Injectable({
  providedIn: 'root'
})
export class LeccionService {
  private baseUrl = `${API_BASE_URL}/api/lecciones`;

  constructor(private http: HttpClient) {}

  getLeccionesByModulo(moduloId: number): Observable<Leccion[]> {
    return this.http.get<Leccion[]>(`${this.baseUrl}/modulo/${moduloId}`);
  }

  getLeccionById(leccionId: number): Observable<Leccion> {
    return this.http.get<Leccion>(`${this.baseUrl}/${leccionId}`);
  }

  createLeccion(leccion: Leccion): Observable<Leccion> {
    return this.http.post<Leccion>(`${this.baseUrl}/register`, leccion);
  }

  updateLeccion(leccionId: number, leccion: Leccion): Observable<Leccion> {
    return this.http.put<Leccion>(`${this.baseUrl}/update/${leccionId}`, leccion);
  }

  deleteLeccion(leccionId: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/delete/${leccionId}`);
  }
}
